contracts/chain-adapters/Arbitrum_Forwarder.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import { ForwarderBase } from "./ForwarderBase.sol";
+import { ArbitrumL1ERC20GatewayLike } from "../interfaces/ArbitrumBridgeInterfaces.sol";
+import { AddressUtils } from "../libraries/AddressUtils.sol";
+
+/**
+ * @title Arbitrum_Forwarder
+ * @notice This contract expects to receive messages and tokens from an authorized contract on a previous layer and forwards them to
+ * contracts on the subsequent layer. It rejects messages which do not originate from a cross domain admin.
+ * @custom:security-contact bugs@across.to
+ */
+contract Arbitrum_Forwarder is ForwarderBase {
+    ArbitrumL1ERC20GatewayLike public immutable arbitrumGatewayRouter;
+
+    modifier onlyFromCrossDomainAdmin() {
```

Comment:
For all the access control functions, there is an exception on L2 only where we have two possible cross domain admins: the hub pool and an upgrade contract. For all forwarders beyond L2, there is only 1 cross domain admin (the previous forwarder).